File: server/src/models/AiChatSession.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export interface IAiChatSession extends Document {
    _id: Types.ObjectId;
    userId: string;
    title: string;
    createdAt: Date;
    updatedAt: Date;
}

const AiChatSessionSchema = new Schema<IAiChatSession>({
    userId: {
        type: String,
        required: true,
        ref: 'User',
        index: true
    },
    title: {
        type: String,
        required: true,
        default: 'New Chat'
    }
}, {
    timestamps: true,
    collection: 'ai_chat_sessions'
});

export const AiChatSession = mongoose.model<IAiChatSession>('AiChatSession', AiChatSessionSchema);

File: splitpal/lib/models/ai_chat_session.dart
class AiChatSession {
  final String id;
  final String title;
  final DateTime updatedAt;

  AiChatSession({
    required this.id,
    required this.title,
    required this.updatedAt,
  });

  factory AiChatSession.fromJson(Map<String, dynamic> json) {
    return AiChatSession(
      id: json['_id'] ?? '',
      title: json['title'] ?? 'New Chat',
      updatedAt: json['updatedAt'] != null 
          ? DateTime.parse(json['updatedAt']) 
          : DateTime.now(),
    );
  }
}
